packages/block-library/src/gallery/edit.js

```diff
@@ -224,6 +229,25 @@ class GalleryEdit extends Component {
 		} );
 	}
 
+	getImagesSizeOptions() {
+		const { imageSizes } = this.props;
+		return map( imageSizes, ( { name, slug } ) => ( { value: slug, label: name } ) );
+	}
+
+	updateImagesSize( sizeSlug ) {
+		const { attributes: { images }, resizedImages } = this.props;
+
+		const updatedImages = map( images, ( image ) => {
+			const url = get( resizedImages, [ parseInt( image.id, 10 ), sizeSlug ] );
```

Comment:
I'm probably not the best person to review this PR but I know that Galleries can have images that are just URLs not coming from the media library. these should be taken into consideration (ignored properly)